### PR TITLE
Add context for client

### LINKED ISSUE #221: Put the kontent client into a react context
### Motivation

It would better fit into the React flow to have the client inside a react context or state. (context avoids props drilling compared to state). As it now stands, once you reset the client, React doesn't ensure that your components get re-rendered with the new client (project id). If the client would be part of the React flow (i.e. in a context or a state) react would re-render your components and ensure you always render the freshest data.

### Proposed solution

Put the Kontent.ai client into a React context instead of having it in a global mutable variable.

### PR DESCRIPTION
**Pull request by IvanKiral**
### Motivation

Which issue does this fix? Fixes #221 

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

**Review by JiriLojda — CHANGES_REQUESTED** [review]
Love the changes. Regarding the renaming of `projectId`s to `environmentId`s, please, replace also occurences of `project`s with `environment`s. There is quite a few of them and I stopped adding comments for each of them, I am sure you can find them. :)

File: src/Client.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  value: string;
+};
+
+const Client = new DeliveryClient({
+  environmentId: currentEnvironmentId,
+  previewApiKey: previewApiKey,
+  defaultQueryConfig: {
+    usePreviewMode: isPreview(),
+  },
+  globalHeaders: (_queryConfig): GlobalHeadersType[] => [
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply src/Client.tsx:67]: I don't understand.. what parameter?

File: src/Client.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        header: sourceTrackingHeaderName,
+        value: `${packageInfo.name};${packageInfo.version}`,
+      },
+    ],
+    propertyNameResolver: camelCasePropertyNameResolver,
+  });
+
+type ClientState = [DeliveryClient, (environmentId: string) => void];
+
+const ClientContext = createContext<ClientState>(undefined as never);
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply src/Client.tsx:94]: as I understand it I would need to have something of ClientState

File: src/Utilities/SelectedProject.ts [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
   IDeliveryNetworkResponse,
   Responses,
 } from '@kontent-ai/delivery-sdk';
 
-const defaultProjectId = '975bf280-fd91-488c-994c-2f04416e5ee3';
+const defaultEnvironmentId = '975bf280-fd91-488c-994c-2f04416e5ee3';
 
 const getSampleProjectItems = (
-  projectId = defaultProjectId
+  environmentId = defaultEnvironmentId
```
[the remark this answers is not in the document]
Comment (IvanKiral) [reply src/Utilities/SelectedProject.ts:11]: forgot about it, deleted

File: src/Client.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (JiriLojda) [reply src/Client.tsx]: The only parameter the lambda accepts. `_queryConfig`

File: src/Client.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (JiriLojda) [reply src/Client.tsx:81]: Oh, right. Then something like `[ClientState, () => {}]`, but its not important, because you always initialize the context.

**Pull request merged**